src/fenics_constitutive/solver.py

```diff
@@ -255,53 +231,44 @@ def form(self, x: PETSc.Vec) -> None:
 
         """
         super().form(x)
-
         assert (
             x.array.data == self._u.vector.array.data
         ), "The solution vector must be the same as the one passed to the MechanicsProblem"
-        if len(self.laws) > 1:
-            for k, (law, cells) in enumerate(self.laws):
-                with df.common.Timer("strain_evaluation"):
-                    # TODO: test this!!
-                    self.del_grad_u_expr.eval(
-                        cells, self._del_grad_u[k].x.array.reshape(cells.size, -1)
-                    )
-
-                with df.common.Timer("stress_evaluation"):
-                    self.submesh_maps[k].map_to_child(self.stress_0, self._stress[k])
-                    if law.history_dim is not None:
-                        self._history_1[0].x.array[:] = self._history_0[0].x.array
-                    law.evaluate(
-                        self._time,
-                        self._del_grad_u[k].x.array,
-                        self._stress[k].x.array,
-                        self._tangent[k].x.array,
-                        self._history_1[k].x.array
-                        if law.history_dim is not None
-                        else None,
-                    )
-
-                with df.common.Timer("stress-local-to-global"):
-                    self.submesh_maps[k].map_to_parent(self._stress[k], self.stress_1)
-                    self.submesh_maps[k].map_to_parent(self._tangent[k], self.tangent)
-        else:
-            law, cells = self.laws[0]
-            with df.common.Timer("strain_evaluation"):
-                self.del_grad_u_expr.eval(
-                    cells, self._del_grad_u[0].x.array.reshape(cells.size, -1)
-                )
 
-            with df.common.Timer("stress_evaluation"):
+        # if len(self.laws) > 1:
+        for k, (law, cells) in enumerate(self.laws):
+            # TODO: test this!
+            # Replace with `self._del_grad_u[k].interpolate(...)` in 0.8.0
+            self.del_grad_u_expr.eval(
+                cells, self._del_grad_u[k].x.array.reshape(cells.size, -1)
+            )
```

Comment:
Are you sure that functions of a QuadratureSpace will have the interpolate functionality?
the fact that these finitle elements do not define any shape functions does not change, right?

Reply:
This has been already merged: https://github.com/FEniCS/dolfinx/pull/2878

And here, `q.interpolate(...)` is used. I think as long as you interpolate a function onto `q` that does not depend on a quadrature space, e.g. `nabla_grad(u)`, then this seems to work. 
https://github.com/FEniCS/dolfinx/blob/6189a7e2f42a63cc2bafd10696862efb41f0c3f9/python/test/unit/fem/test_quadrature_elements.py#L97